src/libfuncs/enum.rs

```diff
@@ -119,11 +120,13 @@ pub fn build_enum_value<'ctx, 'this>(
         helper,
         registry,
         metadata,
-        type_info.variants().unwrap(),
+        type_info
+            .variants()
+            .to_native_assert_error("couldn't get enum's variants")?,
```

Comment:
```suggestion
            .to_native_assert_error("found non-enum type where an enum is required")?,
```
It's not that we can't get an enum's variants because if the type is an enum then they'll always be available. It's rather that the type is not an enum therefore it has no variants.

Reply:
Done!